projects/igniteui-angular/src/lib/calendar/calendar.component.ts

```diff
@@ -996,8 +1129,90 @@ export class IgxCalendarComponent implements OnInit, ControlValueAccessor {
                 this.selectedDates = [start, ...this.generateDateRange(start, end)];
             }
         }
+
+        if (excludeDisabledDates) {
+            this.selectedDates = this.selectedDates.filter(d => !this.isDateDisabled(d));
+        }
+
         this._onChangeCallback(this.selectedDates);
     }
+    /**
+     *@hidden
+     */
+    private selectDateFromClient(value: Date) {
+        switch (this.selection) {
+            case 'single':
+            case 'multi':
+                this.selectDate(value);
+                break;
+            case 'range':
+                this.selectRange(value, true);
+                break;
+        }
+    }
+    /**
+     *@hidden
+     */
+    private isDateInRanges(date: Date, ranges: DateRangeDescriptor[]): boolean {
+        date = new Date(date.getFullYear(), date.getMonth(), date.getDate());
+        const dateInMs = date.getTime();
+
+        for (const descriptor of ranges) {
+            const dRanges = descriptor.dateRange === null ? null :
+                descriptor.dateRange.map(r => new Date(r.getFullYear(),
+                    r.getMonth(), r.getDate()));
+            switch (descriptor.type) {
+                case (DateRangeType.After):
+                    if (dateInMs > dRanges[0].getTime()) {
+                        return true;
+                    }
+
+                    break;
+                case (DateRangeType.Before):
+                    if (dateInMs < dRanges[0].getTime()) {
+                        return true;
+                    }
+
+                    break;
+                case (DateRangeType.Between):
+                    const dRange = dRanges.map(d => d.getTime());
+                    const min = Math.min(dRange[0], dRange[1]);
+                    const max = Math.max(dRange[0], dRange[1]);
+                    if (dateInMs >= min && dateInMs <= max) {
```

Comment:
This would sometimes fail if the provided `min` and `max` dates have a time portion.

Reply:
I strip the time portion of the dates.